File: app/src/main/java/com/dsy/mvp/ui/activity/SplashActivity.java
package com.dsy.mvp.ui.activity;

import android.Manifest;
import android.content.Intent;
import androidx.annotation.NonNull;

import com.dsy.mvp.R;
import com.dsy.mvp.base.BaseActivity;
import com.dsy.mvp.component.PermissionsRequest;
import com.dsy.mvp.presenter.SplashView;
import com.dsy.mvp.presenter.SplashPresenter;

public class SplashActivity extends BaseActivity<SplashPresenter> implements SplashView{

    //TODO 需要申请的权限
    private static final String[] PERMISSIONS = new String[]{
            Manifest.permission.WRITE_EXTERNAL_STORAGE,
//            Manifest.permission.READ_PHONE_STATE
    };
    private PermissionsRequest permissionsRequest;
    @Override
    protected int getLayoutId() {
        return 0;
    }

    @Override
    protected SplashPresenter createPresenter() {
        return new SplashPresenter();
    }

    @Override
    protected void initView() {

        permissionsRequest = new PermissionsRequest(this) {
            @Override
            protected void onPermissionsGranted() {
                toMain();
            }

            @Override
            protected void onPermissionsDenied() {
                finish();
            }
        };
        permissionsRequest.requestPermissions(PERMISSIONS);
    }

    @Override
    protected void initEvent() {
    }

    @Override
    protected void initData() {
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        permissionsRequest.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    @Override
    public void toMain() {
        startActivity(new Intent(getMActivity(),MainActivity.class));
        finish();
    }
}

File: app/src/main/java/com/dsy/mvp/ui/activity/SplashActivity.kt
package com.dsy.mvp.ui.activity

import android.Manifest
import android.content.Intent
import com.dsy.mvp.base.BaseActivity
import com.dsy.mvp.base.SimpleContract
import com.dsy.mvp.base.SimplePresenter
import com.dsy.mvp.component.PermissionsRequest

class SplashActivity : BaseActivity<SimplePresenter>(0) ,SimpleContract.View{

    //TODO 需要申请的权限
    private val permissions = arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE)
    private lateinit var permissionsRequest: PermissionsRequest
    override fun initView() {
        permissionsRequest = object : PermissionsRequest(this) {
            override fun onPermissionsGranted() {
                toMain()
            }

            override fun onPermissionsDenied() {
                finish()
            }
        }
        permissionsRequest.requestPermissions(*permissions)
    }

    override fun initEvent() {}
    override fun initData() {}
    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        permissionsRequest.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    private fun toMain() {
        startActivity(Intent(mActivity, MainActivity::class.java))
        finish()
    }

}